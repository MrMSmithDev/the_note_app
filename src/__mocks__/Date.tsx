import type { DateProps } from '@components/Date/Date';

export default function Date({ date, month, year }: DateProps ) {
  return (
    <div data-testid="mock-date">
      Date for: {`${year}-${month.num}-${date}`}
    </div>
  );
}
