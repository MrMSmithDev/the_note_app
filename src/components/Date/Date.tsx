import Modal from '@components/Modal';
import Note from '@components/Note';
import React, { useRef, useState } from 'react';
import MonthType from 'src/types/MonthType';

interface DateProps {
  date: number;
  month: MonthType;
  year: number;
}

const Date: React.FC<DateProps> = ({ date, month, year }) => {
  const [showModal, setShowModal] = useState<boolean>(false);
  const ref = useRef<HTMLButtonElement>(null);

  function openNote(e: React.MouseEvent<HTMLButtonElement>) {
    e.preventDefault();

    setShowModal(true);
  }

  const dateString = `${year}-${month.num >= 10 ? month.num : '0' + month.num}-${date}`;

  return (
    <div>
      <button
        ref={ref}
        className="cursor-pointer select-none w-full rounded"
        onClick={openNote}
      >
        <span className="text-sm dark:text-gray-400">{date}</span>
      </button>
      {showModal ? (
        <Modal closeModal={() => setShowModal(false)} btnRef={ref}>
          <Note date={dateString} />
        </Modal>
      ) : null}
    </div>
  );
};

export default Date;
