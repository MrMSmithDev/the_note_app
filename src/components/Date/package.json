{
    "name": "date",
    "version": "1.0.0",
    "main": "Date.tsx",
    "dependencies": {
      "react": "^19.0.0"
    }
  }
  