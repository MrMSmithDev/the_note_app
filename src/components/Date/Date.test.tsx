import { render, screen, fireEvent } from '@testing-library/react';
import Date from './Date';

jest.mock('@components/Modal', () => require('@__mocks__/Modal'));
jest.mock('@components/Note', () => require('@__mocks__/Note'));

describe('Date component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly to match snapshot', () => {
    const { asFragment } = render(
      <Date date={3} month={{ abbr: 'MAY', num: 5 }} year={2025} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the date button correctly', () => {
    render(<Date date={5} month={{ abbr: 'FEB', num: 2 }} year={2025} />);

    const btn = screen.getByText('5');
    expect(btn).toBeInTheDocument();
  });

  it('opens the modal when the button is clicked', () => {
    render(<Date date={3} month={{ abbr: 'MAR', num: 3 }} year={2025} />);

    const btn = screen.getByText('3');
    fireEvent.click(btn);

    expect(screen.getByTestId('mock-modal')).toBeInTheDocument();
    expect(screen.getByTestId('mock-note')).toBeInTheDocument();
  });
});
