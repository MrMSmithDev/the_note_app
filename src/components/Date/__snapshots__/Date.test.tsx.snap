// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date component renders correctly to match snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="cursor-pointer select-none w-full rounded hover:bg-gray-200 dark:hover:bg-gray-600"
    >
      <span
        class="text-md md:text-sm dark:text-gray-400"
      >
        3
      </span>
    </button>
  </div>
</DocumentFragment>
`;
